osu.Desktop/DiscordRichPresence.cs

```diff
@@ -92,11 +92,13 @@ private void updateStatus()
                 return;
             }
 
-            if (status.Value == UserStatus.Online && activity.Value != null)
+            if (activity.Value != null)
             {
                 bool hideIdentifiableInformation = privacyMode.Value == DiscordRichPresenceMode.Limited;
+                bool hideInteractions = status.Value == UserStatus.DoNotDisturb && activity.Value is UserActivity.InLobby;
```

Comment:
I do not see why these are separate variables, the intention is that the DnD status should behave *exactly* like "Limited" mode.

Reply:
I thought that was what we agreed we didn't want? I'm confused now. :sweat_smile:

Reply:
You may have misunderstood my [comment](https://github.com/ppy/osu/pull/27354#issuecomment-1965518978), I meant that DnD and "Limited" mode should be the same.

Reply:
Do you mind clarifying why this is your stance over my proposed solution? I honestly find that kind of an extreme solution for the problem in-hand.

Reply:
I wouldn't say that, since the "DnD" status hasn't been given any special behaviour yet, and this looks like one candidate that the player might benefit from (hiding identifiable information when player is in DnD mode, as to not be disturbed).

In addition, "interactions" and "identifiable information" really are one and the same for the existing cases of user statuses, the entire point of the "limited" mode is to prevent the user/player from being disturbed.

I'll try to keep things clear next time before your contribution goes in.

Reply:
Thank you for the clarification, it does help provide some insight.

> I wouldn't say that, since the "DnD" status hasn't been given any special behaviour yet, and this looks like one candidate that the player might benefit from (hiding identifiable information when player is in DnD mode, as to not be disturbed).
> 
> In addition, "interactions" and "identifiable information" really are one and the same for the existing cases of user statuses, the entire point of the "limited" mode is to prevent the user/player from being disturbed.

Hmm, I suppose we have different definitions of "disturbed" then. I see limited mode as more of a privacy feature, as in "I don't want people to see my profile".

I've seen other players get self-conscious and end up playing offline or otherwise hide their profile whether that be from some goofy top plays or them having their real name as their username, so that's what I figured limited mode was for.

In general I tried to think ahead about what DND would probably mean for osu! in the future, and to me I came up with "please don't DM me and please don't join my lobbies, etc." So I suppose that's where that miscommunication originated from.

> I'll try to keep things clear next time before your contribution goes in.

Sweet. It was mostly confusing for me because I was like "wait, why are my changes getting overwritten?" xD

Communication is always good.